core-blocks/paragraph/index.js

```diff
@@ -206,17 +206,16 @@ class ParagraphBlock extends Component {
 		} = attributes;
 
 		const fontSize = this.getFontSize();
-
 		return (
 			<Fragment>
-				<BlockControls>
-					<AlignmentToolbar
+				<MultiBlockControls>
```

Comment:
Does this still support the use-case of the single block selection?

It also prompts me to think: Can we just build-in multi block controls behavior to `<BlockControls />` ?

Reply:
Yes and no, respectively :)

There needs to be some way to show just the multi block controls when multiple blocks are selected. With this approach, it's two slots, one for multi- and one for single- controls. Both are rendered when only one block is selected, and only the multi slot is rendered when we have multiple blocks selected.

I toyed with having some kind of registry, or attribute on elements, but those approaches didn't seem as clean and as obvious to the block developer.

This PR won't work as-is with the current state of master, but I'm really after feedback on the approach at the moment before I go rebasing.

Reply:
>There needs to be some way to show just the multi block controls when multiple blocks are selected. With this approach, it's two slots, one for multi- and one for single- controls. Both are rendered when only one block is selected, and only the multi slot is rendered when we have multiple blocks selected.

So is this just a complexity of the implementation? Or is it conceptually important that, as part of the `edit` interface, a distinction is made between single- and multi- selection?

Reply:
It's a little from column A and a little from column B.

I'd say that allowing the distinction to be made in `edit` allows block developers to specify if they want controls to apply to multiple blocks or not, however, do they need to? Is a block developer ever going to say, "I have this control that can update multiple blocks at once, and I _really_ don't want the user to do that!"

Probably not.

I'd be very happy to hear ideas on how we could flag up controls that should apply to multiple blocks and therefore sort out the rendering purely with `<BlockControls />`. All my previous attempts at that smelled wrong.

Reply:
Seeing the implementation:
```js
const MultiBlockAlignmentToolbar = withMultiBlockSupport( AlignmentToolbar, 'align' );
```
which effectively adds the `attributeName` for the toolbar control behind the scenes. I would bet that it is possible to express all this code as follows:
```js
<BlockControls>
    <BlockAlignmentToolbar
        name="align"
        multi={ true }
        value={ align }
        onChange={ onChange }
    />
</BlockControls>
```
It would require some refactoring for `BlockControls` which is wrapped with `ifBlockEditSelected` HOC. It would have to support also the case when block is multi selected and have `multi` set to `true`. Finally, it would also have to handle the case when `multi` is set to `false` and block is multi selected - not render such control at all.

Still, the biggest challenge would be to iterate over those multiplied controls filled into slot and apply some modifications to the virtual dom tree which would allow to group them, render only one but trigger update for all of them. Basically, replicate what `withMultiBlockSupport` does.

In the future, we could go even further and hide the need for using `value` and `onChange` props since they are usually repeatable so the can be derived from the attribute name as explored in #6705.

Reply:
>In the future, we could go even further and hide the need for using value and onChange props since they are usually repeatable so the can be derived from the attribute name as explored in #6705.

Also #7352